File: cinema-booking-system/backend/src/main/java/com/uit/cinema/booking/service/ReviewService.java
package com.uit.cinema.booking.service;

import com.uit.cinema.booking.dto.request.CreateReviewRequest;
import com.uit.cinema.booking.dto.response.ReviewEligibilityResponse;
import com.uit.cinema.booking.dto.response.ReviewInsightResponse;
import com.uit.cinema.booking.dto.response.ReviewResponse;

import java.util.List;

public interface ReviewService {
    ReviewResponse createReview(CreateReviewRequest request);
    List<ReviewResponse> getMovieReviews(Long movieId);
    List<ReviewResponse> getEventReviews(Long eventId);
    ReviewInsightResponse getMovieInsight(Long movieId);
    ReviewInsightResponse getEventInsight(Long eventId);
    ReviewEligibilityResponse getMovieEligibility(Long userId, Long movieId);
    ReviewEligibilityResponse getEventEligibility(Long userId, Long eventId);
}

File: cinema-booking-system/frontend/src/services/reviewService.ts
import apiClient from './authService';
import type {
  ReviewResponse,
  CreateReviewRequest,
  ReviewInsightResponse,
  ReviewEligibilityResponse,
} from '../types/review';

export const reviewService = {
  /** POST /api/reviews */
  createReview: async (data: CreateReviewRequest): Promise<ReviewResponse> => {
    const response = await apiClient.post<ReviewResponse>('/reviews', data);
    return response.data;
  },

  /** GET /api/reviews/movies/{movieId} */
  getMovieReviews: async (movieId: number): Promise<ReviewResponse[]> => {
    const response = await apiClient.get<ReviewResponse[]>(`/reviews/movies/${movieId}`);
    return response.data;
  },

  /** GET /api/reviews/events/{eventId} */
  getEventReviews: async (eventId: number): Promise<ReviewResponse[]> => {
    const response = await apiClient.get<ReviewResponse[]>(`/reviews/events/${eventId}`);
    return response.data;
  },

  /** GET /api/reviews/movies/{movieId}/insight */
  getMovieInsight: async (movieId: number): Promise<ReviewInsightResponse> => {
    const response = await apiClient.get<ReviewInsightResponse>(`/reviews/movies/${movieId}/insight`);
    return response.data;
  },

  /** GET /api/reviews/events/{eventId}/insight */
  getEventInsight: async (eventId: number): Promise<ReviewInsightResponse> => {
    const response = await apiClient.get<ReviewInsightResponse>(`/reviews/events/${eventId}/insight`);
    return response.data;
  },

  /** GET /api/reviews/movies/{movieId}/eligibility?userId= */
  checkMovieEligibility: async (movieId: number, userId: number): Promise<ReviewEligibilityResponse> => {
    const response = await apiClient.get<ReviewEligibilityResponse>(
      `/reviews/movies/${movieId}/eligibility`,
      { params: { userId } }
    );
    return response.data;
  },

  /** GET /api/reviews/events/{eventId}/eligibility?userId= */
  checkEventEligibility: async (eventId: number, userId: number): Promise<ReviewEligibilityResponse> => {
    const response = await apiClient.get<ReviewEligibilityResponse>(
      `/reviews/events/${eventId}/eligibility`,
      { params: { userId } }
    );
    return response.data;
  },
};
